### DISCUSSION TITLE
[Web] Unexpected null value thrown when focus on editor

### DISCUSSION THREAD

**Discussion by radoslaw-sz** [singerdmx/flutter-quill#197] [Web]
Hello,

Example:

```

import 'package:flutter/material.dart';
import 'package:flutter_quill/models/documents/document.dart';
import 'package:flutter_quill/widgets/controller.dart';
import 'package:flutter_quill/widgets/editor.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatefulWidget {


  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {

  QuillController _controller;

  final FocusNode _focusNode = FocusNode();

  @override
  void initState() {
    final doc = Document()..insert(0, 'Empty asset');
    _controller = QuillController(
      document: doc, selection: TextSelection.collapsed(offset: 0));
    super.initState();
  }
  
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
      body: _quill(),
      )
    );
  }
  Widget _quill() {
    return Column(
      children: <Widget>[
        Container(
          color: Colors.white,
          padding: const EdgeInsets.only(left: 16.0, right: 16.0),
          child: QuillEditor(
            controller: _controller,
            scrollController: ScrollController(),
            scrollable: true,
            focusNode: _focusNode,
            autoFocus: false,
            readOnly: false,
            placeholder: 'Add content',
            enableInteractiveSelection: true,
            expands: false,
            padding: EdgeInsets.zero,
          ),
        ),
      ],
    );
  }
}

```

Makes unexpected null value thrown:

```

══╡ EXCEPTION CAUGHT BY FOUNDATION LIBRARY ╞════════════════════════════════════════════════════════
The following TypeErrorImpl was thrown while dispatching notifications for FocusNode:
Unexpected null value.

When the exception was thrown, this was the stack:
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 236:49      throw_
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 518:63  nullCheck
packages/flutter_quill/widgets/raw_editor.dart 973:25                                                                          [_updateOrDisposeSelectionOverlayIfNeeded]
packages/flutter_quill/widgets/raw_editor.dart 984:5                                                                           [_handleFocusChanged]
packages/flutter/src/foundation/change_notifier.dart 243:16                                                                    notifyListeners
packages/flutter/src/widgets/focus_manager.dart 1052:5                                                                         [_notify]
packages/flutter/src/widgets/focus_manager.dart 1771:11                                                                        [_applyFocusChange]
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/schedule_microtask.dart 40:11                                  _microtaskLoop
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/schedule_microtask.dart 49:5                                   _startMicrotaskLoop
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 166:15               <fn>

The FocusNode sending notification was:
  FocusNode#8a0dc([PRIMARY FOCUS])
════════════════════════════════════════════════════════════════════════════════════════════════════
Another exception was thrown: Unexpected null value.

```

**Comment by IntegroInfotech** [comment]
> Hello,
> 
> Example:
> 
> ```
> 
> import 'package:flutter/material.dart';
> import 'package:flutter_quill/models/documents/document.dart';
> import 'package:flutter_quill/widgets/controller.dart';
> import 'package:flutter_quill/widgets/editor.dart';
> 
> void main() {
>   runApp(MyApp());
> }
> 
> class MyApp extends StatefulWidget {
> 
> 
>   @override
>   _MyAppState createState() => _MyAppState();
> }
> 
> class _MyAppState extends State<MyApp> {
> 
>   QuillController _controller;
> 
>   final FocusNode _focusNode = FocusNode();
> 
>   @override
>   void initState() {
>     final doc = Document()..insert(0, 'Empty asset');
>     _controller = QuillController(
>       document: doc, selection: TextSelection.collapsed(offset: 0));
>     super.initState();
>   }
>   
>   @override
>   Widget build(BuildContext context) {
>     return MaterialApp(
>       home: Scaffold(
>       body: _quill(),
>       )
>     );
>   }
>   Widget _quill() {
>     return Column(
>       children: <Widget>[
>         Container(
>           color: Colors.white,
>           padding: const EdgeInsets.only(left: 16.0, right: 16.0),
>           child: QuillEditor(
>             controller: _controller,
>             scrollController: ScrollController(),
>             scrollable: true,
>             focusNode: _focusNode,
>             autoFocus: false,
>             readOnly: false,
>             placeholder: 'Add content',
>             enableInteractiveSelection: true,
>             expands: false,
>             padding: EdgeInsets.zero,
>           ),
>         ),
>       ],
>     );
>   }
> }
> ```
> 
> Makes unexpected null value thrown:
> 
> ```
> 
> ══╡ EXCEPTION CAUGHT BY FOUNDATION LIBRARY ╞════════════════════════════════════════════════════════
> The following TypeErrorImpl was thrown while dispatching notifications for FocusNode:
> Unexpected null value.
> 
> When the exception was thrown, this was the stack:
> C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 236:49      throw_
> C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 518:63  nullCheck
> packages/flutter_quill/widgets/raw_editor.dart 973:25                                                                          [_updateOrDisposeSelectionOverlayIfNeeded]
> packages/flutter_quill/widgets/raw_editor.dart 984:5                                                                           [_handleFocusChanged]
> packages/flutter/src/foundation/change_notifier.dart 243:16                                                                    notifyListeners
> packages/flutter/src/widgets/focus_manager.dart 1052:5                                                                         [_notify]
> packages/flutter/src/widgets/focus_manager.dart 1771:11                                                                        [_applyFocusChange]
> C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/schedule_microtask.dart 40:11                                  _microtaskLoop
> C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/schedule_microtask.dart 49:5                                   _startMicrotaskLoop
> C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 166:15               <fn>
> 
> The FocusNode sending notification was:
>   FocusNode#8a0dc([PRIMARY FOCUS])
> ════════════════════════════════════════════════════════════════════════════════════════════════════
> Another exception was thrown: Unexpected null value.
> ```

Even I am facing same Problem

**Reply by guruprakashIT** [reply]
Even i am facing the same issue

![image](https://user-images.githubusercontent.com/41444416/119666247-d54b2700-be52-11eb-92ca-ede98974cf0f.png)

**Reply by guruprakashIT** [reply]
pls close this issue because it's work on 

new version 1.3.*
